drivers/rtc/rtc-pcf2127.c

```diff
@@ -685,32 +686,46 @@ static int pcf2127_probe(struct device *dev, struct regmap *regmap,
 	ret = regmap_update_bits(pcf2127->regmap, PCF2127_REG_TS_CTRL,
 				 PCF2127_BIT_TS_CTRL_TSOFF |
 				 PCF2127_BIT_TS_CTRL_TSM,
+				 ts_off ? PCF2127_BIT_TS_CTRL_TSOFF : 0 |
 				 PCF2127_BIT_TS_CTRL_TSM);
 	if (ret) {
 		dev_err(dev, "%s: tamper detection config (ts_ctrl) failed\n",
 			__func__);
 		return ret;
 	}
 
-	/*
-	 * Enable interrupt generation when TSF1 or TSF2 timestamp flags
-	 * are set. Interrupt signal is an open-drain output and can be
-	 * left floating if unused.
-	 */
-	ret = regmap_update_bits(pcf2127->regmap, PCF2127_REG_CTRL2,
-				 PCF2127_BIT_CTRL2_TSIE,
-				 PCF2127_BIT_CTRL2_TSIE);
-	if (ret) {
-		dev_err(dev, "%s: tamper detection config (ctrl2) failed\n",
-			__func__);
-		return ret;
-	}
+	if (!ts_off) {
+		/*
+		 * Enable interrupt generation when TSF1 or TSF2 timestamp flags
+		 * are set. Interrupt signal is an open-drain output and can be
+		 * left floating if unused.
+		 */
+		ret = regmap_update_bits(pcf2127->regmap, PCF2127_REG_CTRL2,
+					 PCF2127_BIT_CTRL2_TSIE,
+					 PCF2127_BIT_CTRL2_TSIE);
+		if (ret) {
+			dev_err(dev, "%s: tamper detection config (ctrl2) failed\n",
+				__func__);
+			return ret;
+		}
 
-	ret = rtc_add_group(pcf2127->rtc, &pcf2127_attr_group);
-	if (ret) {
-		dev_err(dev, "%s: tamper sysfs registering failed\n",
-			__func__);
-		return ret;
+		ret = rtc_add_group(pcf2127->rtc, &pcf2127_attr_group);
+		if (ret) {
+			dev_err(dev, "%s: tamper sysfs registering failed\n",
+				__func__);
+			return ret;
+		}
+	} else {
+		/* Clear timestaping intrrupt flags */
+		ret = regmap_clear_bits(pcf2127->regmap, PCF2127_REG_CTRL1,
+					PCF2127_BIT_CTRL1_TSF1);
+		if (ret)
+			return ret;
+
+		ret = regmap_clear_bits(pcf2127->regmap, PCF2127_REG_CTRL2,
+					PCF2127_BIT_CTRL2_TSF2);
+		if (ret)
+			return ret;
```

Comment:
Hm, isn't it possible to do this in a single `regmap_clear_bits()` statement by logically ORing the two bits together? Or is it required to clear these one by one?

Reply:
These are two different registers.